reloj-checador/app/models/attendace.rb

```diff
@@ -1,3 +1,17 @@
 class Attendace < ApplicationRecord
-    validates :private_number, presence: true
+  validates :private_number, presence: true
+
+  after_create :set_check_type
+
+  def set_check_type
+    attendaces_count = Attendace.where(private_number: self.private_number).count
+    if (attendaces_count > 0 && attendaces_count % 2 == 0)
+      self.check_in = 0
```

Comment:
if you have a single record per checking, rather than one record per work day, then what about having a `check_type` with the values: ['IN', 'OUT']

Or maybe, a single record with:
```
checked_in_at
checked_out_at
```
Then you'd have a single record per workable day with easier reporting

Reply:
Comment taken and changed behavior; Running migration and deleted both fields (check_in and check_out; created instead a new field check_type wich validate and set when a user is entering (IN) o leaving (OUT)